components/azure/azuresynapse/runsparkjob/sample.py

```diff
@@ -0,0 +1,66 @@
+import os
+import kfp.compiler as compiler
+import kfp.components as components
+from kfp.azure import use_azure_secret
+import kfp.dsl as dsl
+import argparse
+
+parser = argparse.ArgumentParser(description='Process inputs.')
+parser.add_argument('--image_name', type=str, default='kubeflow_synapse_component')
+parser.add_argument('--image_repo_name', type=str, default='xiwutest5f386e67')
+args = parser.parse_args()
+
+component_root = os.path.join(os.path.dirname(os.path.abspath(__file__)), ".")
+image_repo_name = args.image_repo_name # the container registery for the container operation and path in the ACR
+image_name = args.image_name
+file_path = os.path.join(component_root, "component.yaml")
+
+# Loading the component.yaml file for deployment operation
+run_job_operation = components.load_component_from_file(file_path)
+
+# The run_job_image_name shall be the container image for the operation
+# It shall be something like <your_acr_name>.azurecr.io/deploy/aml-deploy-model:latest
+run_job_image_name = image_repo_name + '.azurecr.io/deploy/' + image_name + ':latest'
```

Comment:
What if I want to use a local container registry for a locally installed Kubeflow cluster, or I want to use this component with another cloud provider?

It would be best to support any registry, but ideally we provide a prebuilt hosted public image.

Reply:
The component code can run on any registry and any Kubeflow deployment. It doesn't block people build the image, push it to local repo and deploy the pipeline to a local Kubeflow cluster. I added some comments in the sample code and also note in the README.md to call this out.